connector/connect/server/src/main/scala/org/apache/spark/sql/connect/utils/ErrorUtils.scala

```diff
@@ -66,6 +146,18 @@ private[connect] object ErrorUtils extends Logging {
         "classes",
         JsonMethods.compact(JsonMethods.render(allClasses(st.getClass).map(_.getName))))
 
+    if (enrichErrorEnabled) {
+      // Generate a new unique key for this exception.
+      val errorId = UUID.randomUUID().toString
+
+      errorInfo.putMetadata("errorId", errorId)
+
+      SparkConnectService
+        .getOrCreateIsolatedSession(userId, sessionId)
+        .errorIdToError
+        .put(errorId, st)
```

Comment:
THe caller has the session holder, why not pass it instead of calling this again?